### DISCUSSION TITLE
What did 701 Assembly look like?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#112624] [Programming Help]
### Body

Hi,
I am learning X86 Assembly (NASM) and I was curious what 701 Assembly looked like.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by iAmTheBest4222** [comment]
LDX    #LENGTH          LENGTH TO COUNT
         L     A,0              SUM TO ZERO
AGAIN    TD    INPUT,EOF        TEST FOR EOF
         J     EQUAL,EOF        IF SO, DONE
         LDCH  INPUT            GET CHARACTER
         STCH  TEMP             SAVE CHARACTER
         STX   TEMP2            SAVE LENGTH
         L     X,TEMP2          GET LENGTH
         LDCH  TEMP             GET CHARACTER
         ADD   A,0              ADD TO SUM
         J     AGAIN            LOOP UNTIL LENGTH
         H     0                 HALT
EOF      H     0                EOF
TEMP     B     1
TEMP2    B     1
INPUT    B     1
LENGTH   DC    4096

**Comment by ghost** [comment]
What does that code do?
